src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9847,71 +9856,38 @@ private ArgumentSyntax ParseArgumentExpression(bool isIndexer)
                     // as a deconstruction target. I.e. something like "var (x, y)" or "var (x, (y, z))".
                     // We should report an error in this case because we plan to support deconstruction for out arguments
                     // in the future. We need to ensure that we do not successfully interpret this as an invocation of a
-                    // ref-returning method named var with two, or more arguments. 
-                    if (IsDeconstructionCompatibleArgument(refOrOutKeyword, expression))
+                    // ref-returning method named var with two, or more arguments.
+                    if (refOrOutKeyword != null)
                     {
-                        expression = this.AddError(expression, ErrorCode.ERR_OutVarDeconstructionIsNotSupported);
+                        expression = CheckValidLvalue(expression);
                     }
                 }
             }
 
             return _syntaxFactory.Argument(nameColon, refOrOutKeyword, expression);
         }
 
-        private static bool IsDeconstructionCompatibleArgument(SyntaxToken refOrOutKeyword, ExpressionSyntax expression)
+        private ExpressionSyntax CheckValidLvalue(ExpressionSyntax expression)
+        {
+            return IsDeconstructionCompatibleArgument(expression)
+                ? this.AddError(expression, ErrorCode.ERR_VarInvocationLvalueReserved)
+                : expression;
+        }
+
+        private static bool IsDeconstructionCompatibleArgument(ExpressionSyntax expression)
         {
-            if (refOrOutKeyword?.Kind == SyntaxKind.OutKeyword &&
-                expression.Kind == SyntaxKind.InvocationExpression)
+            if (expression.Kind == SyntaxKind.InvocationExpression)
             {
                 var invocation = (InvocationExpressionSyntax)expression;
                 ExpressionSyntax invocationTarget = invocation.Expression;
 
                 return invocationTarget.Kind == SyntaxKind.IdentifierName &&
-                    ((IdentifierNameSyntax)invocationTarget).Identifier.IsVar() &&
-                    invocation.ArgumentList.Arguments.Count > 1 &&
-                    !expression.GetDiagnostics().Contains(info => info.Severity == DiagnosticSeverity.Error) &&
```

Comment:
I'll look at doing this without using Linq in the parser.

---
In reply to: [77872664](https://github.com/dotnet/roslyn/pull/13636#discussion_r77872664) [](ancestors = 77872664)